src/main/scala/mesosphere/chaos/App.scala

```diff
@@ -29,23 +31,34 @@ trait App extends scala.App {
   def run(classes: Class[_ <: Service]*) {
     initConf()
 
+    val services = classes.map { injector.getInstance(_) }
+    val serviceManager = new ServiceManager(services.asJava)
+    this.serviceManager = Some(serviceManager)
+
     sys.addShutdownHook(shutdownAndWait())
 
-    classes.foreach(c => {
-      val service = injector.getInstance(c)
-      services += service
-      service.startAsync()
-    })
+    serviceManager.startAsync()
+
+    Try { serviceManager.awaitHealthy() } match {
+      case Success(_) =>
+        log.info("All services up and running.")
+      case Failure(f) =>
+        log.fatal(s"Failed to start all services. Services by state: ${serviceManager.servicesByState()}", f)
+        shutdownAndWait()
+        System.exit(1)
+    }
   }
 
   def shutdown() {
     log.info("Shutting down services")
-    services.foreach(_.stopAsync())
+    serviceManager.foreach { _.stopAsync() }
```

Comment:
Probably it is just personal taste, so feel free to ignore it but I like parenthesis for small (inline) code blocks without named arguments.